fixtures/xunit_tools.py

```diff
@@ -0,0 +1,252 @@
+            xunit['response']['id']), value=xunit['response']['value'])
+    properties.append(property_resp)
+    for prop_name, prop_value in prop_dict.iteritems():
+        prop_el = etree.Element(
+            'property', name="polarion-{}".format(prop_name), value=str(prop_value))
+        properties.append(prop_el)
+    testsuites.append(properties)
+    testsuites.append(testsuite)
+
+    no_tests = 0
+    results_count = {
+        'passed': 0,
+        'skipped': 0,
+        'failure': 0,
+        'error': 0
+    }
+    for data in tests:
+        no_tests += 1
+        if collectonly:
+            testsuite.append(testresult_gen(data['name'], data['params']))
+            results_count['skipped'] += 1
+        else:
+            testsuite.append(testresult_gen(data['name'], data['params'], result=data['result']))
+            results_count[data['result']] += 1
+    testsuite.attrib['tests'] = str(no_tests)
+    testsuite.attrib['failures'] = str(results_count['failure'])
+    testsuite.attrib['skipped'] = str(results_count['skipped'])
+    testsuite.attrib['errors'] = str(results_count['error'])
+    testsuite.attrib['name'] = "cfme-tests"
+    xml = etree.ElementTree(testsuites)
+    xml.write(filename, pretty_print=True)
+
+
+def get_name(obj):
+    if hasattr(obj, '_param_name'):
+        return getattr(obj, '_param_name')
+    elif hasattr(obj, 'name'):
+        return obj.name
+    else:
+        return str(obj)
+
+
+caselevels = {
+    '0': 'component',
+    '1': 'integration',
+    '2': 'system',
+    '3': 'acceptance'
+}
+
+
+@pytest.mark.trylast
+def pytest_collection_modifyitems(session, config, items):
+    if not config.getoption('generate_xmls'):
+        return
+    a = defaultdict(dict)
+    ntr = []
+    for item in items:
+        a[item.location[0]][re.sub('\[.*\]', '', item.location[2])] = a[item.location[0]].get(
+            re.sub('\[.*\]', '', item.location[2]), 0) + 1
+    with open('duplicates.log', 'w') as f:
+        for module, tests in a.iteritems():
+            for test in tests:
+                if test not in ntr:
+                    ntr.append(test)
+                else:
+                    f.write("{}\n".format(test))
+
+    test_name = []
+
+    testcases = etree.Element("testcases")
+    testcases.attrib['project-id'] = xunit['project_id']
+    response_properties = etree.Element("response-properties")
+    response_property = etree.Element(
+        "response-property", name=xunit['response']['id'], value=xunit['response']['value'])
+    response_properties.append(response_property)
+    properties = etree.Element("properties")
+    lookup = etree.Element("property", name="lookup-method", value="custom")
+    properties.append(lookup)
+    testcases.append(response_properties)
+    testcases.append(properties)
+
+    for item in items:
+        work_items = []
+        custom_fields = {}
+        try:
+            description = item.function.func_doc
+        except:
```

Comment:
- To catch any exception use the ```except Exception```, instead of ```except BaseException``` or just ```except```.
- Indicate the most specific exception type in ```except``` block.

For example:
```python
# Bad
try:
    mapping[key]
except Exception:
    ...

# Better
try:
    mapping[key]
except KeyError:
    ...
```